src/modules/fancyzones/FancyZonesLib/FancyZones.cpp

```diff
@@ -883,20 +889,30 @@ bool FancyZones::OnSnapHotkeyBasedOnZoneNumber(HWND window, DWORD vkCode) noexce
     }
     else
     {
+        auto zoneWindow = m_workAreaHandler.GetWorkArea(m_currentDesktopId, current);
         // Single monitor environment, or combined multi-monitor environment.
         if (m_settings->GetSettings()->restoreSize)
         {
-            bool moved = m_windowMoveHandler.MoveWindowIntoZoneByDirectionAndIndex(window, vkCode, false /* cycle through zones */, m_workAreaHandler.GetWorkArea(m_currentDesktopId, current));
+            bool moved = m_windowMoveHandler.MoveWindowIntoZoneByDirectionAndIndex(window, vkCode, false /* cycle through zones */, zoneWindow);
             if (!moved)
             {
                 FancyZonesUtils::RestoreWindowOrigin(window);
                 FancyZonesUtils::RestoreWindowSize(window);
             }
-            return true;
+            else
+            {
+                Trace::FancyZones::KeyboardSnapWindowToZone(zoneWindow->ActiveZoneSet());
+            }
+            return moved;
```

Comment:
Shouldn't we return `true` here as it was before?

Reply:
I thought here returning moved made more sense, I thought that's what signals success.

Reply:
@SeraphimaZ , What's the idea of the logic here?
I'm assuming the function should return true only if it has moved the windows to a zone, right?

Reply:
Yes, this change makes sense.

Reply:
Thank you. Leaving this change in, then.